Application/ControlMsrE2/ControlMsrE2.c

```diff
@@ -0,0 +1,242 @@
+/** @file
+ Control CFGLock BIOS Option
+ 
+ Copyright (c) 2020, Brumbaer. All rights reserved.<BR>
+ This program and the accompanying materials
+ are licensed and made available under the terms and conditions of the BSD License
+ which accompanies this distribution.  The full text of the license may be found at
+ http://opensource.org/licenses/bsd-license.php
+ 
+ THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+ 
+ **/
+#include "ControlMsrE2.h"
+
+#define CONTEXTS_MAX 8
+
+EFI_STATUS WalkListHeaders (
+                            IN EFI_HII_HANDLE * handles,
+                            IN EFI_HII_PACKAGE_LIST_HEADER **ListHeaders,
+                            IN UINT32 ListHeaderCount,
+                            IN EFI_STRING SearchString
+                            ) {
+    UINT16           OptionsCount;
+    UINT16           ContextsCount;
+    OneOfContext     Contexts[CONTEXTS_MAX];
+    UINT32           ListHeaderIndex;
+    
+    EFI_STATUS Status = EFI_SUCCESS;
+    
+    OptionsCount = 0;
+    ContextsCount = 0;
+    ListHeaderIndex = 0;
+    
+    // For Each Handle
+
+    for (EFI_HII_HANDLE* h = (EFI_HII_HANDLE*) handles; (*h != NULL) && (ContextsCount < CONTEXTS_MAX); h++, ListHeaderIndex++) {
+
+        ListHeaders[ListHeaderIndex] = HiiExportPackageLists(*h);
+        
+        if (ListHeaders[ListHeaderIndex] != NULL) {
+            
+            if (IS_VERBOSE()) {
+                Print (L"Package List: ");
+                PrintGuid (&ListHeaders[ListHeaderIndex]->PackageListGuid);
+                Print (L"\n");
+            }
+            
+            // First package in list
+            EFI_HII_PACKAGE_HEADER* PkgHeader = PADD(ListHeaders[ListHeaderIndex], sizeof(EFI_HII_PACKAGE_LIST_HEADER));
+
+            // For each package in list
+            while (ContextsCount < CONTEXTS_MAX) {
+
+                if (IS_VERBOSE()) {
+                    Print (L"Package Type: %02X ", PkgHeader->Type);
```

Comment:
When dealing with user output it is often more useful to perform printing via `DebugLib`, as it allows to not only present things to user but also save the transcript to the logfile. For user interaction `DEBUG_WARN` level is recommended. I.e. something like this:

```C
  DEBUG ((DEBUG_WARN, "Package Type: %02X ", PkgHeader->Type));
```

The nuance here is that only one-liners are allowed, so if you think this does not fit you, feel free to keep using `Prints` as before.